src/main/java/sorald/sonar/SonarRule.java

```diff
@@ -2,13 +2,14 @@
 
 import java.util.Objects;
 import org.sonarsource.sonarlint.core.rule.extractor.SonarLintRuleDefinition;
+import sorald.rule.IRuleType;
 import sorald.rule.Rule;
 import sorald.rule.RuleType;
 
 public class SonarRule implements Rule {
     private final String key;
     private final String name;
-    private final RuleType type;
+    private final IRuleType type;
```

Comment:
Inside package `sonar`, we can safely use `SonarRuleType`.

Reply:
I would keep programming against the interface instead of the concrete type. This stops a programmer to add special behavior to `SonarRuleType` instead of the `IRuleType` interface.

Reply:
> This stops a programmer to add special behavior to SonarRuleType instead of the IRuleType interface.

Right. We would want to have uniform behaviour across all enums and that is why we are subtyping them with an interface.

Reply:
There is also no real benefit to remove the abstraction and use the concrete class, or? I have more or less learned to program against interfaces, and never really thought about why we do this tbh.

Reply:
> There is also no real benefit to remove the abstraction and use the concrete class, or?

It was readability for me. I could be sure that `type` would be one of the four enums defined in `SonarRuleType`.